Add dropdown to sort completed demons on stats viewers

jasonyess opened the pull request:
An implementation of #208, adds a dropdown which allows the user to choose between two sorting options

![image](https://github.com/user-attachments/assets/a0f8ac81-b273-4178-82c5-f670019d2947)

The selected value is saved in cookies, where the default value of the `simple_dropdown` is set to the cookie value.

In the Javascript (only in the `IndividualStatsViewer` class), the code that populates the stats containers is moved from the `onReceive` function to its own `populateStatsContainers` function, so the stats containers can be repopulated without sending another request

Currently, it's only implemented in the individual stats viewer, but I could see this feature being nice to have in the nation stats viewer as well so I'm open to making changes

I'm not entirely sure if handling all of the cookie stuff in the backend is the greatest idea, my second thought would be to just modify the dropdown's selected value to the cookie value (or maybe just a value in `localStorage`) once it's initialized in the JS, but I'm not quite sure


## License Acceptance

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

stadust commented:
> I'm not entirely sure if handling all of the cookie stuff in the backend is the greatest idea, my second thought would be to just modify the dropdown's selected value to the cookie value (or maybe just a value in localStorage) once it's initialized in the JS, but I'm not quite sure

Mhh, yeah, sorry, suggesting cookies was wrong of me, I meant local storage :sob: I meant localstorage. The tabbed pane stuff (e.g. what allows you to select tabs in the user area and restores your selects tab after reloading) in pointercrate-core-pages/static/js/modules/tab.js already does something like this.

stadust commented:
> Currently, it's only implemented in the individual stats viewer, but I could see this feature being nice to have in the nation stats viewer as well so I'm open to making changes

Let's do it for both, the initialization of the dropdown can be moved into the stats viewer base class, and from there it can probably even re-trigger the abstract population routine. 

I also still think that when its set to "by difficulty", it shouldn't populate everything into the same list, but rather use separate lists for main/extended/legacy demons.

fwiw,

jasonyess commented:
When sorting by difficulty, I implemented the separate lists you talked about
I haven't implemented them to the nation stats viewer yet, but please tell me what I can change

![image](https://github.com/user-attachments/assets/02f5fcea-ced4-404b-b142-dd29a46a8721)
![image](https://github.com/user-attachments/assets/c5847ca5-ac8b-4f25-b9ab-3e213171b8e0)

("None" is replaced with a dash just because it was big compared to the section titles)

In the rust, I also modified the `StatsViewerRow` struct so it's really easy to have multiple sections in one `StatsViewerRow`
![image](https://github.com/user-attachments/assets/d7a26de5-108a-4491-a27b-e75aead4d12e)

stadust commented:
I'll need to have an indepth look at the code tomorrow (looks pretty good at first glance though!), but UI wise I was more thinking of something like this: 
![image](https://github.com/user-attachments/assets/d644d734-6723-4c34-9d31-43573296f687)

jasonyess commented:
oh 😭 i guess i didn't need to modify `StatsViewerRow`, i might as well leave it though since maybe it'll be useful at some point
